flashcards/web/src/modules/AddDeck/components/AddDecksForm.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import { Formik, Form, ErrorMessage } from "formik";
+import * as Yup from "yup";
+import { TextField, Typography, Button, Select, InputLabel, MenuItem, Checkbox, FormControlLabel } from "@mui/material";
+
+const validationSchema = Yup.object({
+    tagName: Yup.string().min(2, "Too Short!").max(20, "Too Long!").required("Required!"),
+    deckName: Yup.string().min(2, "Too Short!").max(20, "Too Long!").required("Required!"),
+    selectTag: Yup.string().required("Required!"),
+    selectCategory: Yup.string().required("Required!"),
+    checkboxPublic: Yup.boolean(),
+    checkboxForSale: Yup.boolean(),
+    price: Yup.number().when("checkboxForSale", {
+        is: true,
+        then: Yup.number().required("Required!"),
+    }),
+});
+
+const initialValues = {
```

Comment:
I'd probably create interface for that first and then implement it.
Most likely values will be of said interface type (line 40)